awx/api/serializers.py

```diff
@@ -1241,6 +1241,19 @@ def get_summary_fields(self, obj):
         return ret
 
 
+class DirectTokenSerializer():
+    def create_token(self, data):
+        if 'token' not in data:
+            data['token'] = generate_token()
+        if 'expires' not in data:
+            data['expires'] = now() + timedelta(
+                seconds=settings.OAUTH2_PROVIDER['ACCESS_TOKEN_EXPIRE_SECONDS']
```

Comment:
Given that this is really intended to be an anonymous, single-use token, I think we should give it a *very* short expiration time, maybe something like 30 minutes (or however long we anticipate the actual sync process should take at maximum).

Reply:
This was something else I thought of but I wasn't sure how long. The default 1000 years is likely too much but I know there are some massive job syncs out there. Do we want to add another setting for this token? PROJECT_SYNC_ACCESS_TOKEN_EXPIRE_SECONDS and start with 24h as the default?

Reply:
repeating from a different conversation - tying this to the launching user is tricky in the case of update-on-launch and scheduled project updates. We'd want to find a way to handle those